Cargo.toml

```diff
@@ -19,68 +19,69 @@ edition = "2021"
-serde_json = { version = "1" }
-serde_tuple = "0.5"
-serde_with = "2.3"
-tempfile = "3.3"
-thiserror = "1"
-tokio = { version = "1", features = ["rt-multi-thread", "macros"] }
-tracing = "0.1"
-tracing-subscriber = "0.3"
-
+anyhow = "1" 
+arbitrary = {version = "1", features = ["derive"]} 
+arbtest = "0.2" 
+async-stm = "0.2" 
+async-trait = "0.1" 
+axum = {version = "0.6", features = ["ws"]} 
+base64 = "0.21" 
+blake2b_simd = "1.0" 
+bytes = "1.4" 
+clap = {version = "4.1", features = ["derive", "env"]} 
+config = "0.13" 
+dirs = "5.0" 
+ethers = {version = "2.0", features = ["abigen", "ws"]} 
+ethers-core = {version = "2.0"} 
+fnv = "1.0" 
+futures = "0.3" 
+hex = "0.4" 
+jsonrpc-v2 = {version = "0.11", default-features = false, features = ["bytes-v10"]} 
+k256 = "0.11" # Same as tendermint-rs
+lazy_static = "1.4" 
+libsecp256k1 = "0.7" 
+multihash = {version = "0.16.1", default-features = false} 
+num-traits = "0.2" 
+paste = "1" 
+prost = {version = "0.11"} 
+quickcheck = "1" 
+quickcheck_macros = "1" 
+rand = "0.8" 
+rand_chacha = "0.3" 
+regex = "1" 
+serde = {version = "1", features = ["derive"]} 
+serde_json = {version = "1"} 
+serde_tuple = "0.5" 
+serde_with = "2.3" 
+tempfile = "3.3" 
+thiserror = "1" 
+tokio = {version = "1", features = ["rt-multi-thread", "macros"]} 
+tracing = "0.1" 
+tracing-subscriber = "0.3" 
 
 # Stable FVM dependencies from crates.io
-fvm = { version = "=3.3", default-features = false }     # no opencl feature or it fails on CI
+fvm = {version = "=3.3", default-features = false}# no opencl feature or it fails on CI
 fvm_ipld_blockstore = "0.1"
-fvm_ipld_encoding = "0.3"
 fvm_ipld_car = "0.6"
+fvm_ipld_encoding = "0.3"
 fvm_ipld_hamt = "0.6"
-fvm_shared = { version = "=3.3", features = ["crypto"] }
+fvm_shared = {version = "=3.3", features = ["crypto"]}
 
 # Using 0.8 because of ref-fvm.
 # 0.9 would be better because of its updated quickcheck dependency.
 # 0.10 breaks some API.
-cid = { version = "0.8", features = ["serde-codec", "std"] }
+cid = {version = "0.8", features = ["serde-codec", "std"]}
 
 # Depending on the release cycle, this dependency might want an earlier version of the FVM.
 # We can work around it by hardcoding the method hashes; currently there is only one.
 # frc42_dispatch = "3.2"
 
 # Using the same tendermint-rs dependency as tower-abci. From both we are interested in v037 modules.
-tower-abci = { version = "0.7" }
-tendermint = { version = "0.31", features = ["secp256k1"] }
-tendermint-rpc = { version = "0.31", features = ["secp256k1", "http-client", "websocket-client"] }
-tendermint-proto = { version = "0.31" }
+tendermint = {version = "0.31", features = ["secp256k1"]}
+tendermint-proto = {version = "0.31"}
+tendermint-rpc = {version = "0.31", features = ["secp256k1", "http-client", "websocket-client"]}
+tower-abci = {version = "0.7"}
+
+[patch.crates-io]
+jsonrpc-v2 = {git = "https://github.com/cryptoAtwill/jsonrpc-v2", branch = "expose_inner_error"}
```

Comment:
@cryptoAtwill, @aakoshh do we wait for Will's patch to be accepted upstream, or should we fork this library in the end?

Reply:
That PR is merged, I think we can switch to upstream.

Reply:
I think we can use the fork until that point.

Reply:
Cool, I bumped the version in this PR to pick up the latest one.